[BUG] GHA can fail because nuget fails to verify downloaded packages

**What is the bug?**
Running `dotnet restore` during GitHub Action fails with errors like
```
/home/runner/work/opensearch-net/opensearch-net/tests/Tests.Configuration/Tests.Configuration.csproj : error NU3028: Package 'runtime.osx.10.10-x64.runtime.native.System.Security.Cryptography.Apple 4.3.0' from source 'https://api.nuget.org/v3/index.json': The repository primary signature's timestamp found a chain building issue: UntrustedRoot: self signed certificate in certificate chain [/home/runner/work/opensearch-net/opensearch-net/OpenSearch.sln]
```

See [here](https://github.com/opensearch-project/opensearch-net/runs/6925714704?check_suite_focus=true#step:6:11) for a full example.

**What is the expected behavior?**
Workflows on GitHub to succeed.

**What is your host/environment?**
Building using GitHub Actions only..


**Additional Information**
This is related to https://github.com/NuGet/Announcements/issues/56

--- a/.github/workflows/compile.yml
+++ b/.github/workflows/compile.yml
@@ -10,7 +10,7 @@ jobs:
     - uses: actions/checkout@v2
     - uses: actions/setup-dotnet@v1
       with:
-        dotnet-version: '5.0.100'
+        dotnet-version: '5.0.405'
     - uses: actions/cache@v2
       with:
         path: ~/.nuget/packages

--- a/.github/workflows/integration.yml
+++ b/.github/workflows/integration.yml
@@ -43,7 +43,7 @@ jobs:
         uses: actions/checkout@v2
       - uses: actions/setup-dotnet@v1
         with:
-          dotnet-version: '5.0.100'
+          dotnet-version: '5.0.405'
       - uses: actions/cache@v2
         with:
           path: ~/.nuget/packages
@@ -76,7 +76,7 @@ jobs:
         uses: actions/checkout@v2
       - uses: actions/setup-dotnet@v1
         with:
-          dotnet-version: '5.0.100'
+          dotnet-version: '5.0.405'
       - uses: actions/cache@v2
         with:
           path: ~/.nuget/packages

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -12,7 +12,7 @@ jobs:
     - uses: actions/checkout@v2
     - uses: actions/setup-dotnet@v1
       with:
-        dotnet-version: '5.0.100'
+        dotnet-version: '5.0.405'
     - uses: actions/cache@v2
       with:
         path: ~/.nuget/packages

--- a/.github/workflows/test-jobs.yml
+++ b/.github/workflows/test-jobs.yml
@@ -26,7 +26,7 @@ jobs:
         uses: actions/checkout@v2
       - uses: actions/setup-dotnet@v1
         with:
-          dotnet-version: '5.0.100'
+          dotnet-version: '5.0.405'
       - uses: actions/cache@v2
         with:
           path: ~/.nuget/packages

--- a/.github/workflows/test-jobs.yml
+++ b/.github/workflows/test-jobs.yml
@@ -66,7 +66,7 @@ jobs:
         uses: actions/checkout@v2
       - uses: actions/setup-dotnet@v1
         with:
-          dotnet-version: '5.0.100'
+          dotnet-version: '5.0.405'
       - uses: actions/cache@v2
         with:
           path: ~/.nuget/packages

--- a/global.json
+++ b/global.json
@@ -1,6 +1,6 @@
 {
   "sdk": {
-    "version": "5.0.100",
+    "version": "5.0.405",
     "rollForward": "latestFeature",
     "allowPrerelease": false
   },

